src/core/manager.c

```diff
@@ -4245,6 +4245,11 @@ ManagerState manager_state(Manager *m) {
 
         assert(m);
 
+        /* Is the special shutdown target active or queued? If so, we are in shutdown state */
+        u = manager_get_unit(m, SPECIAL_SHUTDOWN_TARGET);
+        if (u && unit_active_or_pending(u))
+                return MANAGER_STOPPING;
+
```

Comment:
Hmm, this change looks reasonable. But it needs to be a separate commit with an appropriate title and an explanation what issue it is solving and when that issue occurs.

Reply:
This has been done with PR #15330